docs/x509.rst

```diff
@@ -652,6 +652,33 @@ X.509 Extensions
     purposes indicated in the key usage extension. The object is
     iterable to obtain the list of :ref:`extended key usage OIDs <eku_oids>`.
 
+.. class:: AuthorityKeyIdentifier
+
+    .. versionadded:: 0.9
+
+    The authority key identifier extension provides a means of identifying the
+    public key corresponding to the private key used to sign a certificate.
+
+    .. attribute:: key_identifier
+
+        :type: bytes
+
+        A value derived from the public key used to verify the certificate's
+        signature. See :rfc:`5280` section 4.2.1.2 for more details on the
+        recommended way to compute this value.
```

Comment:
That, plus a link to the place in the RFC that explains it should be ok.

On Sun, Apr 19, 2015 at 9:55 AM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1835#discussion_r28653010>:
>
> > @@ -652,6 +652,33 @@ X.509 Extensions
> >      purposes indicated in the key usage extension. The object is
> >      iterable to obtain the list of :ref:`extended key usage OIDs <eku_oids>`.
> >
> > +.. class:: AuthorityKeyIdentifier
> > +
> > +    .. versionadded:: 0.9
> > +
> > +    The authority key identifier extension provides a means of identifying the
> > +    public key corresponding to the private key used to sign a certificate.
> > +
> > +    .. attribute:: key_identifier
> > +
> > +        :type: bytes
> > +
> > +        A value derived from the public key used to verify the certificate's
> > +        signature. See :rfc:`5280` section 4.2.1.2 for more details on the
> > +        recommended way to compute this value.
>
> I think explaining the semantics of how a certificate chain may be built
> is way too complex to fit in this section. We have links to normative
> references where things (like this) can't be simplified. "What should I do
> if I parsed a certificate with some value" here is basically "you shouldn't
> touch this, it's not for you".
>
> Maybe simply stating "this extension is typically used by software to
> assist in determining the appropriate certificate chain" would be
> sufficient?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1835/files#r28653010>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084